internal/translator/data/rules.go

```diff
@@ -34,6 +36,106 @@ type Rules struct {
 // RulesHelper has useful functions to work with the collection
 type RulesHelper interface {
 	FindByPattern(ctx context.Context, source, target, pattern string) ([]Rule, error)
+	NewRule(ctx context.Context, ruleStruct Rule) (*primitive.ObjectID, error)
+	FindRuleByPattern(ctx context.Context, sourceLanguage, targetLanguage, pattern string) ([]Rule, error)
+	UpdateOne(ctx context.Context, filter Rule, updateValue Rule) error
+	DeleteOne(ctx context.Context, rule Rule) error
```

Comment:
`DeleteOne` usually receive the ObjectID

Reply:
the function receives and interface object, not an ObjectID like here teach.

https://www.mongodb.com/blog/post/quick-start-golang--mongodb--how-to-delete-documents

Let me know how you do it with ObjectID because we don't do that in other files

Reply:
Instead of receive a reference to a `Rule` struct. You need to receive an `primitive.ObjectID`

So you can make a struct
```
map[string]interface{}{
"_id": objectID
}
```
And that's it, because usually when you are trying to delete one document what you have is its id.

Reply:
Does it make sense to you?

Reply:
yes, very clear, thanks